src/Axe/Managers/CompilerManager.cs

```diff
@@ -265,10 +276,105 @@ public CompilerManager(AxeDbContext context, INodeServices nodeServices) : base(
+        {
+            try
+            {
+                CodeBlock codeBlock = context.CodeBlock
+                    .Include(cb => cb.TestCases)
+                    .First(cb => cb.Id == model.Id);
+                List<string> failedTestCases = new List<string>();
+
+                string pyFilePath = Path.GetTempPath() + Path.GetRandomFileName().Replace(".", "") + ".py";
+
+                //ProcessStartInfo startInfo = new ProcessStartInfo("C:\\Users\\andrei.shilkin\\AppData\\Local\\Programs\\Python\\Python36-32\\python.exe");
+                ProcessStartInfo startInfo = new ProcessStartInfo(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + "\\AppData\\Local\\Programs\\Python\\Python36-32\\python.exe");
+                startInfo.Arguments = pyFilePath;
+                startInfo.UseShellExecute = false;
+                startInfo.CreateNoWindow = true;
+                startInfo.RedirectStandardOutput = true;
+                startInfo.RedirectStandardError = true;
+
+                for (int i = 0; i < codeBlock.TestCases.Count; i++)
+                {
+                    // Generate the code for execution
+                    string code = model.SourceCode;
+                    if (codeBlock.TestCases[i].Input != null)
+                    {
+                        string[] inputArguments = codeBlock.TestCases[i].Input.Split(' ');
+                        code += "\nprint(main(";
+                        for (int j = 0; j < inputArguments.Length; j++)
+                        {
+                            code += inputArguments[j];
+                            if (j < inputArguments.Length - 1)
+                            {
+                                code += ", ";
+                            }
+                        }
+                        code += "));";
+                    }
+                    else
+                    {
+                        code += "\nprint(main());";
+                    }
+                    // Write the code into temporary file
+                    using (var pyFile = File.Create(pyFilePath))
+                    {
+                        using (var stream = new StreamWriter(pyFile))
+                        {
+                            stream.Write(code);
+                        }
+                    }
+                    // Create process for Python and execute script
+                    Process process = new Process();
+                    process.StartInfo = startInfo;
+                    process.Start();
+                    // Read data from terminal
+                    string error;
+                    using (var stream = process.StandardError)
+                    {
+                        error = await stream.ReadToEndAsync();
+                    }
+                    if (error.Length > 0)
+                    {
+                        return new Tuple<CodeBlockResult, string[]>(CodeBlockResult.Error, new string[] { error });
```

Comment:
It is time to create specialized class to replace all those `Tuple<CodeBlockResult, string[]>`